packages/salesforcedx-vscode-apex/src/languageServer.ts

```diff
@@ -102,19 +101,14 @@ export function setupDB(): void {
     vscode.workspace.workspaceFolders &&
     vscode.workspace.workspaceFolders[0]
   ) {
-    const dbPath = path.join(
-      vscode.workspace.workspaceFolders[0].uri.fsPath,
-      '.sfdx',
-      'tools',
-      'apex.db'
-    );
-    if (fs.existsSync(dbPath)) {
+    const dbPath = projectPaths.getApexLanguageServerDatabasePath();
+    if (dbPath && fs.existsSync(dbPath)) {
```

Comment:
TS beefed about this being potentially undefined, hence the additional conditionals.

Reply:
Should our projectPaths functions never return undefined?  Throwing an error if conditions aren't met to build the path seems like it'd be fine.

Reply:
This block is taking action only if there is already a path to the Apex language server database. So it checks and then if there is a path, it tries to delete the existing resource. If we changed it to throw when it tried to get the path to the database, then we would have to wrap this in a try catch and specifically handle the thrown exception, and then continue with execution. Seems to be like it would be more complicated, but open to further discussion.

Reply:
hum...so apexLanguageServerDatabase only has the undefined type if getRootWorkspace returns undefined. I don't think we should ever get to this point if we don't have a root workspace so shouldn't we be ok to throw in that case and remove the undefined return type.

Reply:
Ok yeah that makes sense.  I updated it.